### PR TITLE
feat: add modal

### PR DESCRIPTION
**Pull request by Rekaterina**

### CODE REVIEW

File: src/components/App/App.tsx [hunk 1]
```diff
@@ -29,6 +35,12 @@ export default class App extends React.Component<{}, AppState> {
             selectedSorting: SortOption.ReleaseDate,
             movies,
             selectedMovieId: null,
+            showModal: true,
+
+            // change isEditMovieModal/isAddMovieModal/isDeleteMovieModal to see use cases
+            isEditMovieModal: true,
```
Comment (Vight86) [inline src/components/App/App.tsx:41]: Optional. it is better in such cases not to add Boolean for each type of dialog. It is better to have one state variable. For example movieModalType.
For this in TS we even can use typing like type ModalType = 'Edit' | 'Delete' | 'Add'

File: src/components/App/App.tsx [hunk 2]
```diff
@@ -52,13 +64,55 @@ export default class App extends React.Component<{}, AppState> {
...
+    };
+
+    renderModal() {
+        if (!this.state.showModal) return null;
+        let dialogContent = null;
+        let dialogTitle = '';
+
+        if (this.state.isAddMovieModal) {
+            dialogContent = <MovieForm onSubmit={this.onFormSubmit} />;
+            dialogTitle = 'Add Movie';
```
Comment (Vight86) [inline src/components/App/App.tsx:90]: Then here we can use map. For example 
`const DIALOG_TITLE_MAP = {
  Add: 'Add Movie',
  Delete: 'Delete movie'
}`
And use it like 
`
  dialogContent = DIALOG_MOVIE_MAP[this.state.movieModalType]
`

File: src/components/App/App.tsx [hunk 3]
```diff
@@ -52,13 +64,55 @@ export default class App extends React.Component<{}, AppState> {
...
+        } else if (this.state.isEditMovieModal) {
+            dialogContent = <MovieForm initialMovie={movies[0]} onSubmit={this.onFormSubmit} />;
```
Comment (Vight86) [inline src/components/App/App.tsx:92]: And for dialogContent we can use like
<MovieForm onSubmit={this.onFormSubmit}  {...(this.state.movieModalType === "Edit" ? {initialMovies: movies[0]} : {})}  />.
Or we can jus map prop config for movie form in component

File: src/components/App/App.tsx (hunk 2 above)
Comment (Rekaterina) [reply src/components/App/App.tsx]: fixed

**Pull request merged**